velox/functions/prestosql/DecimalVectorFunctions.cpp

```diff
@@ -22,6 +22,19 @@
 namespace facebook::velox::functions {
 namespace {
 
+struct ExtraParams {
+  ExtraParams() {
+    params.round.decimal.reset();
```

Comment:
ASAN is complaining about `params.round.decimal` not initialized in this case.  Maybe left all the constructors out (they do not play well with `union`) and just zero-initialize the values when you create it `ExtraParams extraParams{}`, or change `std::optional<int32_t>` to `int32_t` and use -1 to represent absent.

Reply:
@karteekmurthys You need to remove the constructors otherwise `ExtraParams{}` still calls the constructor

Reply:
Thanks for the suggestion. I am trying this ExtraParams extraParams{}. Lmk if this still fails ASAN @Yuhta.

Reply:
@karteekmurthys Can you remove constructor definitions at line 26, 30, 32?